### ISSUE TITLE
Facebook Instagram App Rejected, Due to improper handling of media file CDN

### ISSUE THREAD

**Issue by shivamkb17** [chatwoot/chatwoot#8498]
### Describe the bug

For long time facebook app is rejected, I am testing it recently, I didn't get URL, I see media file, while mentioning anyone on the instagram user. 

For Story mention, Facebook CDN is working properly. 
But when I see the media file shared by user, Chatwoot is saving that media file in the active storage. 

Facebook need Facebook CDN based url for all media types in instagram messaging. 

This is the rejection reply from meta team. 

During our review, we found that your bot does not comply with the quality guidelines and other specific requirements for Instagram Messaging in our technical documentation. Specifically, proper handling of media CDN URLs by not storing/caching the media content.
To resolve this issue, update your app to follow our technical documentation, then re-submit for review.
For more details on submission requirements please read our [Messenger Platform documentation](https://developers.facebook.com/docs/messenger-platform/overview) and our app review [documentation for Instagram Messaging](https://developers.facebook.com/docs/messenger-platform/instagram/app-review).
Notes from your reviewer:
We found that your app does not comply with the quality guidelines and other specific requirements for Instagram Messaging in our technical documentation.

To resolve this issue, visit our Instagram Messaging documentation and ensure that your app meets our requirements, including:

- Proper implementation of Story @mention and Story Reply flows.

- Proper handling of media CDN URLs by not storing/caching the media content.

For more details on submission requirements please visit our app review documentation for Instagram Messaging: https://developers.facebook.com/docs/messenger-platform/instagram/app-review

### To Reproduce

Facebook App approval rejected. 

Submit facebook app submission. 

### Expected behavior

Expected behaviour, Chatwoot doesn't need to save the media file into active storage. 

Media File will also handled with facebook CDN. 

### Environment

Linux VM

### Cloud Provider

GCP

### Platform

None

### Operating system

Ubuntu 20.04 LTS

**Comment by shivamkb17** [comment]
Currenly CDN handling media file only for story mention and story reply , but now instagram reviewer need CDN based URL for media file shared in the instagram inbox.

**Comment by shivamkb17** [comment]
This is update from facebook direct Support. 
The app is not following the requirement 3. 

Thank you for contacting Facebook Direct Support.

Sorry for the inconvenience caused to you. Please note that for the app ID (728454xxxxxxxxxx), as per the review team, the instagram_manage_messages permission was rejected as the app fails to comply with technical guidelines (CDN).

Please note that the review team also found that your application does not comply with the quality guidelines and other specific requirements for Instagram Messaging in the technical documentation. To resolve this issue, visit the Instagram Messaging documentation and ensure that your app meets our requirements, including: 
1. The ability to send and receive Instagram Messages with an escalation path to a human agent.  
2. Proper implementation of Story @mention and Story Reply flows. 
3. Proper handling of media CDN URLs by not storing/caching the media content.

Please note that, as per the review team, they mentioned that app fails to comply with technical guidelines (CDN). So, please note that, as per the CDN URL policy, when you send the image from Instagram to the application, you need to open the image in the application, and after opening the image in the application, the URL of the image should show related to Instagram.

For more details on submission requirements, please visit our app review documentation for Instagram Messaging: https://developers.facebook.com/docs/messenger-platform/instagram/app-review

Answering your question, please note that application may work properly only once the permission is approved. We appreciate you for mentioning the problem you are facing.

We apologise for the inconvenience you are facing. Please note that the apps in development mode can only receive test notifications initiated through the app dashboard or notifications initiated by people who have a role on the app.

For more information about meta webhooks, please access the below link. https://developers.facebook.com/docs/graph-api/webhooks/

We kindly appreciate you for asking a question in the developer forum. Also, please note that you can file a bug report regarding the issue you are facing.

Please note that you can access the below link to file a bug report. https://developers.facebook.com/support/bugs/

If you have any further questions about the app review process, please do not hesitate to reach out to us.

Best regards, Filipe Moiduson | Facebook Direct Support |

**Comment by sojan-official** [comment]
closing in favor of https://github.com/chatwoot/chatwoot/issues/8512#issuecomment-1919197676 
We will track this under one ticket
